model/historical_test_data.go

```diff
@@ -1,39 +1,51 @@
 package model
 
 import (
-	"bytes"
 	"context"
-	"io/ioutil"
-	"path/filepath"
+	"crypto/sha1"
+	"fmt"
+	"hash"
+	"io"
 	"time"
 
 	"github.com/evergreen-ci/cedar"
-	"github.com/evergreen-ci/pail"
+	"github.com/mongodb/anser/bsonutil"
+	"github.com/mongodb/anser/db"
 	"github.com/mongodb/grip"
 	"github.com/mongodb/grip/message"
 	"github.com/pkg/errors"
 	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
+const historicalTestDataCollection = "historical_test_data"
+
 // HistoricalTestData describes aggregated test result data for a given date
 // range.
 type HistoricalTestData struct {
-	Info            HistoricalTestDataInfo `bson:"-"`
+	ID              string                 `bson:"_id,omitempty"`
+	Info            HistoricalTestDataInfo `bson:"info"`
 	NumPass         int                    `bson:"num_pass"`
 	NumFail         int                    `bson:"num_fail"`
-	Durations       []time.Duration        `bson:"durations"`
-	AverageDuration time.Duration          `bson:"average_duration"`
+	AverageDuration float64                `bson:"average_duration"`
```

Comment:
Would it be too hard to store it as a `time.Duration` instead so we know it's nanoseconds and have a native Go time type? It might reduce the amount of rounding error that will result each time we add a data point. It seems like this could be done by just removing the `.Seconds()` rounding in the update pipeline.

Reply:
The problem with storing it as `time.Duration` is that we are doing the calculation in mongo, so it gets stored as a `float64`, and decoding fails when we want to go back to `time.Duration`.

Reply:
What if we keep it as a `float64` but store it as nanoseconds in the database, and convert to seconds in `rest/model/historical_test_data.go`?

Reply:
I updated the code to store ns in the database.

Reply:
Can we unmarshal it as `time.Duration` then? It seems like if it's already in nanoseconds, it could be converted to the correct type (from float to long) by using [$toLong](https://docs.mongodb.com/manual/reference/operator/aggregation/toLong).

Reply:
Done.